### PR TITLE
Select tweets

### PR DESCRIPTION
**Pull request by CrossNox**

### CODE REVIEW

File: question.py [hunk 1]
```diff
@@ -72,19 +69,43 @@ def create_tags(app_uuid, user_id, tags):
...
                    score=tag["tag"])
         all_tags = list(Tag.select().where(Tag.text_uuid == tag["text_uuid"]))
-        if len(all_tags) >= MIN_QUESTIONS:
-            all_scores = [t.score for t in all_tags]
-            # It is tempting to use standard deviation, but it is not sensitive
-            # to outliers. e.g. stdev([1, -0.5, -0.6]) ~= 0.9
-            if min_diff(all_scores) <= MAX_DIFF:
+
+        # I shall only consider texts tagged more than MIN_TAGS times
+        # as candidadtes for complete texts
```
Comment (Roj) [inline question.py]: candidadtes => candidates

File: question.py [hunk 2]
```diff
@@ -72,19 +69,43 @@ def create_tags(app_uuid, user_id, tags):
...
+        # Whether they are completed depends on the error margin given by those 
+        # tags
+        if len(all_tags) >= MIN_TAGS:
+            mean = statistics.mean([tag.score for tag in all_tags])
+            sigma = statistics.stdev([tag.score for tag in all_tags])
+            error = 1.96 * sigma / sqrt(len(all_tags))
```
Comment (Roj) [inline question.py]: Why 1.96?

File: question.py [hunk 3]
```diff
@@ -1,23 +1,31 @@
...
 from jsmin import jsmin
 
+import random
+
 # TODO: put in a .yml file
 NUM_QUESTIONS = 3
 MIN_QUESTIONS = 2 # Integrity constraint for Text.completed
-MAX_DIFF = 0.4 # between any two elements (invariant to outliers)
+MIN_TAGS = 5
+ERROR_TRESHOLD = 0.05
```
Comment (Roj) [inline question.py]: ERROR_THRESHOLD

File: question.py [hunk 4]
```diff
@@ -72,19 +69,43 @@ def create_tags(app_uuid, user_id, tags):
...
+            return False
+        if not text.completed:
+            continue
+
+        all_tags = list(Tag.select().where(Tag.text_uuid == tag["text_uuid"]))
+        mean = statistics.mean([tag.score for tag in all_tags])
+        sigma = statistics.stdev([tag.score for tag in all_tags])
+        error = 1.96 * sigma / sqrt(len(all_tags))
+        LOGGER.debug("info values {}, {}".format(mean, sigma))
+        if not (mean - sigma <= float(tag["tag"]) and float(tag["tag"]) <= mean + sigma):
```
Comment (Roj) [inline question.py]: fun fact: python allows a <= b <= c

**Review by Roj — CHANGES_REQUESTED** [review]

File: question.py (hunk 2 above)
Comment (CrossNox) [reply question.py]: https://en.wikipedia.org/wiki/1.96
I'm still changing this to use a t-student distribution (It's supposed to be better for small samples)

**Review by Roj — APPROVED** [review]

**Pull request merged**